addons/mail/models/discuss/discuss_channel.py

```diff
@@ -687,12 +688,14 @@ def _notify_get_recipients(self, message, msg_vals=False, **kwargs):
                   FROM res_partner partner
              LEFT JOIN res_users users on partner.id = users.partner_id
                  WHERE partner.active IS TRUE
-                       AND partner.email != %s
-                       AND partner.id = ANY(%s) AND partner.id != ANY(%s)"""
-            self.env.cr.execute(
-                sql_query,
-                (email_from or '', list(pids), [author_id] if author_id else [], )
```

Comment:
Note: this is a bugfix, this part was not working as expected when author_id was False (guest in channel in particular) it returned no results.